UI/src/view/municipalityWorkList.js

```diff
@@ -89,13 +89,73 @@
           me.reloadForm(municipalityId, true);
         });
 
+      var privateRoadInfoListButton = $('<a>').addClass('btn btn-primary btn-private-road-list')
+        .text('Yksityistiet').attr("href", "#work-list/privateRoads/" + municipalityId);
+
+      var downloadPDF = function() {
+      pdf.css('height', '1700px').css('width', '700px').addClass('pdfBackgroundColor');
+        var html = $('#formTable').html();
+
+        var takeOutElements = function(beginElement, endElement, sizeElement){
+          while (html.indexOf(beginElement) != -1) {
+            var l = html.length;
+            var i = html.indexOf(beginElement);
+            var e = html.indexOf(endElement, i);
+            var s = html.slice(i, e + sizeElement);
+            html = html.replace(s, '');
+          }
+        };
+        takeOutElements('<button', '</button>', 9);
+        takeOutElements('<a', '</a>', 4);
+        takeOutElements('<th>', '</th>', 5);
+        takeOutElements('<td><input', '</td>', 5);
+        takeOutElements('<th id="suggestedAssets">', '</th>', 5);
+        takeOutElements('<td headers="suggestedAssets">', '</td>', 5);
+        takeOutElements('<div id="test">', '</div>', 6);
+
+        pdf.html('<div class="work-list">' + '<div id="formTable">' + html + '</div>'+ '</div>');
+
+        var HTML_Width = $("#pdf").width();
+        var HTML_Height = $("#pdf").height();
+        var top_left_margin = -5;
+        var PDF_Width = HTML_Width+(top_left_margin*2);
+        var PDF_Height = 985;
+        var canvas_image_width = HTML_Width;
+        var canvas_image_height = HTML_Height;
+        var totalPDFPages = Math.ceil(HTML_Height/PDF_Height)-1;
+
+        html2canvas(document.querySelector('#pdf')).then( function (canvas){
+              var imgData = canvas.toDataURL('image/jpeg');
+          var doc = new jsPDF('p', 'pt', [PDF_Width, PDF_Height]);
+          doc.addImage(imgData, 'JPEG', top_left_margin, top_left_margin,canvas_image_width,canvas_image_height);
+
+          for (var i = 1; i <= totalPDFPages; i++) {
+            doc.addPage(PDF_Width, PDF_Height);
```

Comment:
Since this line is not change, could be move for outside of the loop

Reply:
Even though the loop doesn't change it directly by the counter it is still necessary to be there since the pages are variable (depends on the quantity of assets) and so the totalPDFPages dictate them